### PR TITLE
Code cleanups

### PR DESCRIPTION
**Pull request by vuntz**
Simplify some helper methods by avoiding unneeded blocks.

This should address https://github.com/crowbar/crowbar-ha/pull/69#discussion_r50835031

cc @jsuchome

### CODE REVIEW

File: crowbar_framework/app/models/pacemaker_service_object.rb [hunk 1]
```diff
@@ -97,22 +97,18 @@ def is_remotes?(element)
...
     end
 
     # Returns: name of the barclamp and of the proposal for this cluster
     def cluster_get_barclamp_and_proposal(element)
-      if is_cluster?(element) || is_remotes?(element)
-        ["pacemaker", cluster_name(element)]
-      else
-        [nil, nil]
-      end
+      return [nil, nil] unless is_cluster?(element) || is_remotes?(element)
```
Comment (jsuchome) [inline crowbar_framework/app/models/pacemaker_service_object.rb]: `&&` ??

Comment (vuntz) [inline crowbar_framework/app/models/pacemaker_service_object.rb]: Good catch!

File: chef/cookbooks/crowbar-pacemaker/libraries/helpers.rb [hunk 2]
```diff
@@ -108,15 +96,13 @@ def self.cluster_haproxy_vadmin_name(node)
...
-        vpublic_name
-      end
-    else
+    return nil unless cluster_enabled?(node)
+
+    vpublic_name = node[:pacemaker][:haproxy][:public_name]
+    if vpublic_name.nil? || vpublic_name.empty?
       nil
+    else
+      vpublic_name
```
Comment (sayalilunkad) [inline chef/cookbooks/crowbar-pacemaker/libraries/helpers.rb]: I guess you can also change this part to something like this:
return nil if vpublic_name.nil? || vpublic_name.empty?
vpublic_name

File: crowbar_framework/app/models/pacemaker_service_object.rb (hunk 1 above)
Comment (vuntz) [inline crowbar_framework/app/models/pacemaker_service_object.rb]: Actually no, current code is correct. I think :-)

File: chef/cookbooks/crowbar-pacemaker/libraries/helpers.rb (hunk 2 above)
Comment (vuntz) [inline chef/cookbooks/crowbar-pacemaker/libraries/helpers.rb]: Yes, that'd be an option. I only wanted to move away the "is not a cluster" cases, though. Want me to do the change?

File: crowbar_framework/app/models/pacemaker_service_object.rb (hunk 1 above)
Comment (jsuchome) [inline crowbar_framework/app/models/pacemaker_service_object.rb]: Ouch, yes, you are right and I can't read logical expressions... :-(

File: chef/cookbooks/crowbar-pacemaker/libraries/helpers.rb (hunk 2 above)
Comment (sayalilunkad) [inline chef/cookbooks/crowbar-pacemaker/libraries/helpers.rb]: I have no strong opinion. Just thought it would look cleaner this way :)

**Comment by jsuchome** [comment]
+1 (no opinion about that remainng issue)

**Comment by aplanas** [comment]
Gating fail not related (Ports in binding_failed: 1)

**Comment by vuntz** [comment]
We're hitting https://trello.com/c/30w2SCA0/128-c6-ha-waiting-for-testvm-booted-and-ping-returned-timed-out-in-testsetup-step :/ Tempest passed though, so I'll consider this good enough for merging.

**Pull request merged**